core/src/main/java/cz/xtf/core/namespace/NamespaceManager.java

```diff
@@ -0,0 +1,194 @@
+        return isCreated;
+    }
+
+    /**
+     * Deletes namespace as returned by @see #getNamespace
+     *
+     * @param waitForDeletion whether to wait for deletion (timeout 2 min)
+     * 
+     * @return true if successful, false otherwise
+     */
+    public static boolean deleteProject(boolean waitForDeletion) {
+        return deleteProject(getNamespace(), waitForDeletion);
+    }
+
+    /**
+     * Deletes namespace
+     *
+     * @param namespace namespace name to delete
+     * @param waitForDeletion whether to wait for deletion (timeout 2 min)
+     *
+     * @return true if successful, false otherwise
+     */
+    public static boolean deleteProject(String namespace, boolean waitForDeletion) {
+        boolean deleted = false;
+        OpenShift openShift = OpenShifts.master(namespace);
+        if (openShift.getProject() != null) {
+            deleted = openShift.deleteProject();
+            log.info("Start deleting namespace: " + openShift.getNamespace());
+            if (!deleted && waitForDeletion) {
+                BooleanSupplier bs = () -> openShift.getProject() == null;
+                new SimpleWaiter(bs, TimeUnit.MINUTES, 2, "Waiting for " + openShift.getNamespace() + " project deletion")
+                        .waitFor();
+            }
+        }
+        return deleted;
+    }
+
+    private static String getNamespaceForTestClass(TestDescriptor testDescriptor) {
+        if (OpenShiftConfig.useNamespacePerTestcase()) {
+            // some test case names can be really long resulting in long namespace names. This can cause issues
+            // with routes which have 64 chars limit for prefix of domain name. In case of route like:
+            // galleon-provisioning-xml-prio-mnovak-galleonprovisioningxmltest.apps.eapqe-024-dryf.eapqe.psi.redhat.com
+            // route prefix is: galleon-provisioning-xml-prio-mnovak-galleonprovisioningxmltest
+            // route suffix is: .apps.eapqe-024-dryf.eapqe.psi.redhat.com
+            if ((XTFConfig.get(OPENSHIFT_NAMESPACE) + "-"
+                    + testDescriptor.getParent().get().getDisplayName().toLowerCase())
+                            .length() > OpenShiftConfig.getNamespaceLengthLimitForUniqueNamespacePerTest()) {
+
+                return XTFConfig.get(OPENSHIFT_NAMESPACE) + "-"
+                        + StringUtils.truncate(DigestUtils.sha256Hex(testDescriptor.getParent().get().getDisplayName()
+                                .toLowerCase()),
+                                OpenShiftConfig.getNamespaceLengthLimitForUniqueNamespacePerTest()
+                                        - XTFConfig.get(OPENSHIFT_NAMESPACE).length());
+            } else {
+                return XTFConfig.get(OPENSHIFT_NAMESPACE) + "-"
+                        + testDescriptor.getParent().get().getDisplayName().toLowerCase(); // namespace must not have upper case letters
+            }
+        } else {
+            return XTFConfig.get(OPENSHIFT_NAMESPACE);
```

Comment:
`OpenShiftConfig.getNamespace()` can be used?

Reply:
fixed